src/Symfony/Component/Form/Metadata/Loader/AttributeLoader.php

```diff
@@ -0,0 +1,140 @@
+            }
+
+            $options = $typeAttribute->getOptions();
+            if (null === $viewName = $typeAttribute->getName()) {
+                $fieldName = $name;
+            } else {
+                $options['property_path'] = $name;
+                $fieldName = $viewName;
+            }
+
+            $fields[$fieldName] = new FieldMetadata($fieldName, $typeAttribute->getType(), $options);
+        }
+
+        return new FormMetadata(
+            $class,
+            $this->closestAsFormTypeParent($reflectionClass),
+            $fields,
+            $asFormType->getOptions(),
+        );
+    }
+
+    public function support(string $class): bool
+    {
+        return class_exists($class) && $this->isAsFormType(new \ReflectionClass($class));
+    }
+
+    /**
+     * @return iterable<string, Type>
+     */
+    private function getFields(\ReflectionClass $reflectionClass): iterable
+    {
+        foreach ($reflectionClass->getProperties() as $propRef) {
+            if (
+                $reflectionClass->getName() !== $propRef->getDeclaringClass()->getName()
+                || null === $typeAttribute = $this->getOneAttributeInstance(Type::class, $propRef)
+            ) {
+                continue;
+            }
+
+            yield $propRef->getName() => $typeAttribute;
+        }
+    }
+
+    /**
+     * @param class-string<T> $attributeClass
+     *
+     * @return T|null
+     */
+    private function getOneAttributeInstance(string $attributeClass, \ReflectionProperty|\ReflectionClass $ref): ?object
+    {
+        foreach ($this->getAttributeInstances($attributeClass, $ref) as $attrInstance) {
+            return $attrInstance;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param class-string<T> $attributeClass
+     *
+     * @return iterable<T>
+     */
+    private function getAttributeInstances(string $attributeClass, \ReflectionProperty|\ReflectionClass $ref): iterable
+    {
+        /** @var \ReflectionAttribute<T> $attrRef */
+        foreach ($ref->getAttributes() as $attrRef) {
+            if (is_a($attrRef->getName(), $attributeClass, true)) {
+                yield $attrRef->newInstance();
+            }
+        }
+    }
+
+    private function closestAsFormTypeParent(\ReflectionClass $reflectionClass): ?string
+    {
+        while ($parent = $reflectionClass->getParentClass()) {
+            if ($this->isAsFormType($parent)) {
+                return $parent->getName();
+            }
+
+            $reflectionClass = $parent;
+        }
+
+        return null;
+    }
+
+    private function isAsFormType(\ReflectionClass $reflectionClass): bool
+    {
+        foreach ($reflectionClass->getAttributes() as $attrRef) {
+            if (is_a($attrRef->getName(), AsFormType::class, true)) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
The attributes can be filtered when calling `ReflectionClass::getAttributes()`.

```suggestion
        return (bool) $reflectionClass->getAttributes(AsFormType::class, \ ReflectionAttribute::IS_INSTANCEOF);
```